src/network/live_configuration/live_configuration.rs

```diff
@@ -0,0 +1,310 @@
+// (C) Copyright IBM Corp. 2025.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::{Arc, Mutex};
+
+use super::current_mode::CurrentModeOfflineReason;
+use super::update_thread_worker::UpdateThreadWorker;
+use super::{CurrentMode, Error, Result};
+use crate::client::configuration::Configuration;
+use crate::network::http_client::ServerClient;
+use crate::utils::{ThreadHandle, ThreadStatus};
+use crate::ConfigurationId;
+
+pub trait LiveConfiguration {
+    /// Returns the current configuration
+    ///
+    /// Depending on the current operation mode (see [`LiveConfiguration::get_current_mode`]) and
+    /// the configured offline behavior (see [`OfflineMode`]) for this object, this
+    /// configuration might come from different sources: server, cache or user-provided.
```

Comment:
The current implementation for this only returns the configuration in Online mode and errors otherwise.
Is doc not correct here or is this still missing?